auslib/admin/views/history.py

```diff
@@ -1,96 +1,150 @@
-import difflib
 import json
+import time
 
-from sqlalchemy.sql.expression import null
+from flask import Response, jsonify, request
+from sqlalchemy import and_
 
-from flask import Response
-
-from auslib.global_state import dbo
 from auslib.admin.views.base import AdminView
+from auslib.util.timesince import timesince
+
+
+class HistoryAdminView(AdminView):
 
+    history_keys = ('timestamp', 'change_id', 'data_version', 'changed_by')
 
-class FieldView(AdminView):
-    """/view/:type/:id/:field"""
-
-    def get_value(self, type_, change_id, field):
-        tables = {
-            'rule': dbo.rules,
-            'permission': dbo.permissions,
-            'release': dbo.releases,
-        }
-        if type_ not in tables:
-            raise KeyError('Bad table')
-        table = tables[type_]
-        revision = table.history.getChange(change_id=change_id)
-        if not revision:
-            raise ValueError('Bad change_id')
-        if field not in revision:
-            raise KeyError('Bad field')
-        return revision[field]
-
-    def format_value(self, value):
-        if isinstance(value, dict):
-            try:
-                value = json.dumps(value, indent=2, sort_keys=True)
-            except ValueError:
-                pass
-        elif value is None:
-            value = 'NULL'
-        elif isinstance(value, int):
-            value = unicode(str(value), 'utf8')
-        else:
-            value = unicode(value, 'utf8')
-        return value
-
-    def get(self, type_, change_id, field):
+    def getAllRevisionKeys(self, revisions, primary_keys):
```

Comment:
I'm just starting the review. 
I had a free time at  work and I started, now I'm going home, haha.
I still have to put the comments on history.py, check the not used code and squash the commits.